specs/eventing/overview.md

```diff
@@ -1,97 +1,162 @@
+  events flow along a [channel](#channel) to all
+  [subscriptions](#subscription)). This model can be thought of as "event
+  plumbing" in that events are managed like flows of water through pipes. In
+  order to enable event processing, subscriptions provide a mechanism to route
+  "reply" events further through the object topology.
+
+- Content-based event routing ([`eventing.knative.dev`](#eventing))
+
+  Events are selected for routing based on the event _attributes_ rather than
+  primarily by object connections (a [broker](#broker) provides a stream of
+  events which can be selected by a [trigger](#trigger)). This model is more
+  akin to picking parts off a conveyor belt, where each event is considered
+  separately for processing. Content-based routing handles reply events by
+  re-enqueuing the events in the originating Broker.
+
+![Overview of objects](images/eventing-overview.svg)
+
+<!-- Generated from the following UML using PlantUML:
+@startuml
+' Remove shadows
+skinparam shadowing false
+' Set ordering
+left to right direction
+
+usecase Addressable <<Knative Service>>
+usecase Addressable AS A2 <<Kubernetes Service>>
+
+package "eventing.knative.dev" {
+agent Broker
+agent T1 [
+Trigger
+--
+events selected from Broker,
+replies sent back to Broker
+]
+agent "Trigger" as T2
+}
+
+Broker <-- T1 : broker
+Broker <-- T2 : broker
+Addressable <-u- T1 : subscriber
+
+package "messaging.knative.dev" {
+agent Channel
+agent "Subscription" as S1
+agent S2 [
+Subscription
 ---
-
-_Navigation_:
-
-- [Motivation and goals](motivation.md)
-- **Resource type overview**
-- [Interface contracts](interfaces.md)
-- [Object model specification](spec.md)
+events sent from Channel to Addressable,
+replies sent to second Addressable
+]
+}
+
+Channel <-- S1 :channel
+Channel <-- S2 :channel
+Addressable <-u- S2 :subscriber
+A2 <-u- S2 : reply
+@enduml
+-->
+
+Knative Eventing does not directly specify mechanisms for other event-processing
+models, including multi-stage workflows, correlated request-reply, and
+sequential (windowed) event processing; these models could be built using the
+primitives provided by Knative, or Knative could deliver events to an external
+system that implements these models.
+
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
+
+## Interface Contracts
+
+In addition to the concrete types described below in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the referenced resources must support. The
+following interface contracts define resource fragments and partial schemas
+(required fields on an arbitrary API object) which form a basis for Knative
```

Comment:
I'm not sure how to interpret the "required" here. This doc isn't normative so I guess we could ignore it, however, is this trying to say that all Sinks MUST be duck-type "Addressable" ? If so, what does it mean then for K8s Services to not be? Or are we treating that as an extension by the reference impl? If so, we should probably make it clear (some place) that it's ok to extend things in that way.

Reply:
K8s services are _not_ Addressable, but they are valid targets for a Destination, as a special case. See lines 105-108.

We explicitly call out Destination as having a special case for K8s Services, because the mapping from a K8s Service to its DNS name is not reported in `status`.

Reply:
right and that's my point. Statements like "required fields on an arbitrary API object" isn't a true statement when we think about K8s Service. Either way, I'd like to make this a point of extensibility since I think it's unreasonable to assume that all other systems we integrate with can be modified to change their `status` section (or in some cases, add one). I'm not a fan of spec authors doing things that implementer of the spec can not - and so us decided that K8s Service is special but some other resource types can't leverage that same special-ness doesn't make for a good spec. It's begging for an extensibility point.